pkg/apis/application/v1alpha1/types_test.go

```diff
@@ -2592,3 +2596,45 @@ func Test_validateGroupName(t *testing.T) {
 		})
 	}
 }
+
+func TestGetCAPath(t *testing.T) {
+
+	temppath, err := ioutil.TempDir("", "argocd-cert-test")
+	if err != nil {
+		panic(err)
+	}
```

Comment:
I think we should make a differentiation between the actual code path tested (use `assert` and `require`), and unrelated, only test-specific prerequisites failing. This is why I used `panic`, so that we have a clear separation between a test failing because the code is bad, or a test failing because a test pre-requisite is failing.

Reply:
Interesting, I'd always thought of `require` as being designed for prereqs. But panic seems reasonable for that as well. 

Aside: there's a `t.TempDir()` that handles failures and cleanup for you. But that can wait for a different PR.